HPXMLtoOpenStudio/tests/test_hvac_sizing.rb

```diff
@@ -621,6 +622,122 @@ def test_allow_increased_fixed_capacities
+        clg_cap_orig = hpxml_bldg.heat_pumps[0].cooling_capacity
+        backup_htg_cap_orig = hpxml_bldg.heat_pumps[0].backup_heating_capacity
+        # apply autosizing factor
+        hpxml, hpxml_bldg = _create_hpxml('base-hvac-air-to-air-heat-pump-1-speed.xml')
+        hpxml_bldg.heat_pumps[0].backup_heating_capacity = nil
+        hpxml_bldg.heat_pumps[0].heating_capacity = nil
+        hpxml_bldg.heat_pumps[0].cooling_capacity = nil
+        hpxml_bldg.heat_pumps[0].heating_autosizing_factor = haf
+        # use a reverse factor for backup heating sizing
+        hpxml_bldg.heat_pumps[0].backup_heating_autosizing_factor = (2.0 - haf)
+        hpxml_bldg.heat_pumps[0].cooling_autosizing_factor = caf
+        XMLHelper.write_file(hpxml.to_doc, @tmp_hpxml_path)
+        _model, _hpxml, hpxml_bldg = _test_measure(args_hash)
+        assert_in_epsilon(hpxml_bldg.heat_pumps[0].heating_capacity, htg_cap_orig * haf, 0.001)
+        assert_in_epsilon(hpxml_bldg.heat_pumps[0].backup_heating_capacity, backup_htg_cap_orig * (2.0 - haf), 0.001)
+        assert_in_epsilon(hpxml_bldg.heat_pumps[0].cooling_capacity, clg_cap_orig * caf, 0.001)
+
+        # Test heat pump w/ detailed performance
+        hpxml, hpxml_bldg = _create_hpxml('base-hvac-air-to-air-heat-pump-var-speed-detailed-performance-autosize.xml')
+        hpxml_bldg.heat_pumps[0].backup_heating_capacity = nil
+        XMLHelper.write_file(hpxml.to_doc, @tmp_hpxml_path)
+        _model, _hpxml, hpxml_bldg = _test_measure(args_hash)
+        htg_cap_orig = hpxml_bldg.heat_pumps[0].heating_capacity
+        clg_cap_orig = hpxml_bldg.heat_pumps[0].cooling_capacity
+        backup_htg_cap_orig = hpxml_bldg.heat_pumps[0].backup_heating_capacity
+        # apply autosizing factor
+        hpxml, hpxml_bldg = _create_hpxml('base-hvac-air-to-air-heat-pump-var-speed-detailed-performance-autosize.xml')
+        hpxml_bldg.heat_pumps[0].backup_heating_capacity = nil
+        hpxml_bldg.heat_pumps[0].heating_capacity = nil
+        hpxml_bldg.heat_pumps[0].cooling_capacity = nil
+        hpxml_bldg.heat_pumps[0].heating_autosizing_factor = haf
+        hpxml_bldg.heat_pumps[0].cooling_autosizing_factor = caf
+        # use a reverse factor for backup heating sizing
+        hpxml_bldg.heat_pumps[0].backup_heating_autosizing_factor = (2.0 - haf)
+        XMLHelper.write_file(hpxml.to_doc, @tmp_hpxml_path)
+        _model, _hpxml, hpxml_bldg = _test_measure(args_hash)
+        assert_in_epsilon(hpxml_bldg.heat_pumps[0].heating_capacity, htg_cap_orig * haf, 0.001)
+        assert_in_epsilon(hpxml_bldg.heat_pumps[0].backup_heating_capacity, backup_htg_cap_orig * (2.0 - haf), 0.001)
+        assert_in_epsilon(hpxml_bldg.heat_pumps[0].cooling_capacity, clg_cap_orig * caf, 0.001)
+
+        # Test allow fixed capacity
+        hpxml, hpxml_bldg = _create_hpxml('base-hvac-undersized.xml')
+        hpxml_bldg.header.allow_increased_fixed_capacities = true
```

Comment:
Interesting test. If I'm reading the `hvac_sizing.rb` code correctly, in this situation we calculate the autosized capacity, then apply the factor, and finally check if it still needs to be increased. Do I have that right? That does seem like the desired behavior.

Reply:
Exactly! 👍